### PR TITLE
Update CouchbaseList.Remove to update list on the server

### PR DESCRIPTION
**Pull request by wattsm**
This PR updates the `Remove` method of `CouchbaseList` so that it removes the item from the list on the server. 

Changes in this PR:
- Expanded tests for `CouchbaseList<T>`
- Abstract item retrieval in `CouchbaseList<T>` into `GetItems`
- Update `Remove` to fetch items from the server, find the index of the item to be removed and then use `IBucket.Remove(string path)` to remove it by index.

~PS. I cannot currently login to Gerrit to sign the CLA, after clicking sign-in and authorizing with GitHub I get a screen which just says "Not Found".~ The link from the bot below worked :+1:

### CODE REVIEW

File: Src/Couchbase/Collections/CouchbaseList.cs [hunk 1]
```diff
@@ -197,21 +153,48 @@ private T Get(int index)
...
+            return items[index];
+        }
+
+        private List<T> GetItems() 
+        {
             // ReSharper disable once InconsistentlySynchronizedField
             var get = Bucket.Get<List<T>>(Key);
 
-            if (!get.Success)
+            if(!get.Success)
```
Comment (MikeGoldsmith) [inline Src/Couchbase/Collections/CouchbaseList.cs]: Looks like some whitespace got edited here accidentally.

**Comment by jeffrymorris** [comment]
@MikeGoldsmith @wattsm - I don't see this in Gerrit (code review system).  It may be that your email doesn't in Gerrit doesn't match the email used by the commit. @wattsm can you confirm and also make sure you have signed the CLA?

**Comment by wattsm** [comment]
Hi @jeffrymorris - I've signed the CLA in Gerrit, but in my settings I have no email address and I cannot see how I add one?

I'll fix the whitespace shortly.

**Comment by MikeGoldsmith** [comment]
Hi @wattsm - thanks for updating to fix the whitespace.
The only other change we need is to update the commit message to follow our format. I can do that once we have the change in Gerrit. Please follow the SDK-Bot instructions above to sign the CLA and as Jeff indicates, you need to make sure the email address you signed the commits with matches with Gerrit.

I created the JIRA issue [NCBC-1699](https://issues.couchbase.com/browse/NCBC-1699) to track this change.

Thanks

**Comment by wattsm** [comment]
Hi @MikeGoldsmith - I've got an account on Gerrit and signed the CLA, however my account did not have an email address. I've tried to add one but have not received the validation email yet.

**Comment by MikeGoldsmith** [comment]
I'll see if I can prod it to send you the confirmation, or find someone who can.

**Comment by MikeGoldsmith** [comment]
Hi @wattsm - I'm really sorry for the delay. My usual contact for Gerrit admin is on vacation. I'll keep prodding to make sure your permissions are fixed

In the mean time, I'm going to pull your commit and upload on your behalf directly into Gerrit to get this PR moving forward. I'll share the link when it's done.

**Comment by wattsm** [comment]
Cool, thanks @MikeGoldsmith

**Comment by MikeGoldsmith** [comment]
Gerrit link: http://review.couchbase.org/#/c/95416/

**Comment by MikeGoldsmith** [comment]
Hi @wattsm - FYI I've accepted your change (you can view it on Gerrit). Thanks again for your contribution.

Please can you close this PR? It lost the connection between this and your change in Gerrit.

**Comment by wattsm** [comment]
Nice one, thanks @MikeGoldsmith 👍

**Pull request closed without merging**